self-core-impl/src/main/java/com/selfxdsd/core/BitbucketWebhooks.java

```diff
@@ -0,0 +1,200 @@
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+ * OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.selfxdsd.core;
+
+import com.selfxdsd.api.Project;
+import com.selfxdsd.api.Webhook;
+import com.selfxdsd.api.Webhooks;
+import com.selfxdsd.api.storage.Storage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+import javax.json.JsonValue;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Bitbucket repo webhooks.
+ *
+ * @author Ali FELLAHI (fellahi.ali@gmail.com)
+ * @version $Id$
+ * @since 0.0.68
+ *
+ * @todo #1015:60min Continue writing tests for BitbucketWebhooks class.
+ */
+final class BitbucketWebhooks implements Webhooks {
+
+    /**
+     * Logger.
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(
+        BitbucketWebhooks.class
+    );
+
+    /**
+     * Bitbucket repo Webhooks base uri.
+     */
+    private final URI hooksUri;
+
+    /**
+     * Gitlab's JSON Resources.
+     */
+    private final JsonResources resources;
+
+    /**
+     * Self storage, in case we want to store something.
+     */
+    private final Storage storage;
+
+    /**
+     * Ctor.
+     *
+     * @param resources Gitlab's JSON Resources.
+     * @param hooksUri Hooks base URI.
+     * @param storage Storage.
+     */
+    BitbucketWebhooks(final JsonResources resources,
+                      final URI hooksUri,
+                      final Storage storage) {
+        this.resources = resources;
+        this.hooksUri = hooksUri;
+        this.storage = storage;
+    }
+
+    @Override
+    public boolean add(final Project project) {
+        LOG.debug("Adding Bitbucket webhook for Project "
+            + project.repoFullName());
+        final boolean added;
+        final Resource response = this.resources.post(
+            this.hooksUri,
+            Json.createObjectBuilder()
+                .add("description", "Self-XDSD PM")
+                .add("url", System.getenv(Env.WEBHOOK_BASE_URL)
+                    + "/bitbucket/" + project.repoFullName())
+                .add("active", true)
+                .add("events",
+                    Json.createArrayBuilder()
+                        .add("repo:push")
+                        .add("issue:created")
+                        .add("issue:comment_created")
+                        .add("pullrequest:created")
+                        .add("pullrequest:comment_created")
+                )
```

Comment:
@amihaiemil I'm not sure if these are all the events that we are interested in. 
The list of all available events is:
```
pullrequest:unapproved
issue:comment_created
repo:imported
repo:created
repo:commit_comment_created
pullrequest:approved
pullrequest:comment_updated
issue:updated
project:updated
repo:deleted
pullrequest:changes_request_created
pullrequest:comment_created
repo:commit_status_updated
pullrequest:updated
issue:created
repo:fork
pullrequest:comment_deleted
repo:commit_status_created
repo:updated
pullrequest:rejected
pullrequest:fulfilled
pullrequest:created
pullrequest:changes_request_removed
repo:transfer
repo:push
```
as described here https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Bworkspace%7D/%7Brepo_slug%7D/hooks